src/ast/node/function.rs

```diff
@@ -307,6 +297,98 @@ impl Node for FuncCallNode {
     }
 }
 
+pub fn generic_tp_apply<'a, 'b, T: Generic + CustomType, N: GenericInferenceAble + RangeTrait>(
+    t: &T,
+    n: &N,
+    ctx: &'b mut Ctx<'a>,
+    builder: &'b BuilderEnum<'a, '_>,
+) -> Result<(), PLDiag> {
+    *ctx.need_highlight.borrow_mut() += 1;
+    let generic_size = t.get_generic_size();
+    let generic_params = n.get_generic_params();
+    let range = n.range();
+    let mut generic_types = preprocess_generics(generic_params, generic_size, range, ctx, builder)?;
+
+    let re = ctx.run_in_type_mod(t, |ctx, t| {
+        ctx.protect_generic_context(t.get_generic_map(), |ctx| {
+            for (i, (_, pltype)) in t.get_generic_map().iter().enumerate() {
+                if i >= generic_size {
+                    break;
+                }
+                if generic_types[i].is_none() {
+                    // fill inferred types
+                    let ty = n
+                        .get_inference_result()
+                        .clone()
+                        .unwrap_or_default()
+                        .get(i)
+                        .map(|v| {
+                            let ty = ctx.unify_table.borrow_mut().probe(*v);
+                            ty.get_type(ctx, builder, &mut ctx.unify_table.clone().borrow_mut())
+                        });
+                    if let Some(ty) = ty {
+                        if *ty.borrow() != PLType::Unknown {
+                            generic_types[i] = Some(ty);
+                        } else {
+                            continue;
+                        }
+                    } else {
+                        continue;
+                    }
+                }
+                let res = ctx.eq(pltype.clone(), generic_types[i].as_ref().unwrap().clone());
```

Comment:
[Question]: what is this for? I don't know why we need to compare the equality of `pltype` and `generic_types`. 

- `generic_types`: infered by `n`, which is the argument
- `pltype`: it is the generic list of a function

Does it check the correctness of the inference?

Reply:
Yes, this is the old logic of inference, witch could check the correctness of our new inference method, and doing more stuffs like trait constraints check etc.